src/vmm/src/vmm_config/net.rs

```diff
@@ -61,45 +61,25 @@ pub struct NetworkInterfaceUpdateConfig {
 }
 
 /// Errors associated with `NetworkInterfaceConfig`.
-#[derive(Debug, derive_more::From)]
+#[derive(Debug, derive_more::From, thiserror::Error)]
 pub enum NetworkInterfaceError {
-    /// Could not create Network Device.
+    /// Could not create Network Device
+    #[error("Could not create Network Device: {0}")]
     CreateNetworkDevice(devices::virtio::net::Error),
-    /// Failed to create a `RateLimiter` object.
+    /// Failed to create a `RateLimiter` object
+    #[error("Failed to create a `RateLimiter` object: {0}")]
     CreateRateLimiter(std::io::Error),
-    /// The MAC address is already in use.
+    /// The MAC address is already in use
+    #[error("The MAC address is already in use: {0}")]
     GuestMacAddressInUse(String),
-    /// Error during interface update (patch).
+    /// Error during interface update (patch)
+    #[error("Error during interface update (patch): {0}")]
     DeviceUpdate(VmmError),
-    /// Cannot open/create tap device.
+    /// Cannot open/create tap device
+    #[error("Cannot open/create tap device: {0}")]
     OpenTap(TapError),
 }
```

Comment:
If you are deriving `thiserror::Error` you don't need `derive_more::From` you can simply place `#[from]` in front of the variants to you want to derive from for.
```suggestion
#[derive(Debug, thiserror::Error)]
pub enum NetworkInterfaceError {
    /// Could not create Network Device.
    #[error("Could not create Network Device: {0}")]
    CreateNetworkDevice(#[from] devices::virtio::net::Error),
    /// Failed to create a `RateLimiter` object.
    #[error("Failed to create a `RateLimiter` object: {0}")]
    CreateRateLimiter(#[from] std::io::Error),
    /// The MAC address is already in use.
    #[error("The MAC address is already in use: {0}")]
    GuestMacAddressInUse(String),
    /// Error during interface update (patch).
    #[error("Error during interface update (patch): {0}")]
    DeviceUpdate(#[from] VmmError),
    /// Cannot open/create tap device.
    #[error("Cannot open/create tap device: {0}")]
    OpenTap(#[from] TapError),
}
```

Reply:
Yup, forgot about that feature, added now